### PR TITLE
Allow downgrades using RPM and DEB packages

### PR DESCRIPTION
**Pull request by jr0me**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/22522|

## Description

This PR modifies RPM and DEB scripts for Wazuh agent's packages.

The modifications make it possible to downgrade Wazuh to earlier versions, allowing to rollback upgrades or to undo transactions via package managers across more versions.

## Tests

- Build the package in any supported platform
  - [x] Linux
- [x] Package installation
- [x] Package upgrade
- [x] Package downgrade
- [x] Package remove
- [x] Package install/remove/install
- [x] Change added to CHANGELOG.md

### CODE REVIEW

File: debs/SPECS/wazuh-agent/debian/postrm [hunk 1]
```diff
@@ -51,6 +51,37 @@ case "$1" in
     ;;
 
     upgrade)
+        # If the upgrade downgrades to earlier versions, restore ownership
+        if command -v ${DIR}/bin/ossec-control > /dev/null 2>&1; then
+
+            OSMYSHELL="/sbin/nologin"
+
+            if [ -f ${DIR}/queue/sockets/.agent_info ]; then
+                mv ${DIR}/queue/sockets/.agent_info ${DIR}/queue/ossec/
```
Comment (TomasTurina) [inline debs/SPECS/wazuh-agent/debian/postrm:60]: Would also be necessary to remove the `sockets` folder?

File: debs/SPECS/wazuh-agent/debian/postrm [hunk 2]
```diff
@@ -51,6 +51,37 @@ case "$1" in
...
+
+            if ! getent group ossec > /dev/null 2>&1; then
+                addgroup --system ossec > /dev/null 2>&1
+            fi
+
+            if ! getent passwd ossec > /dev/null 2>&1; then
+                adduser --system --home /var/ossec --shell ${OSMYSHELL} --ingroup ossec ossec > /dev/null 2>&1
+            fi
+
+            find ${DIR} ! -group root -exec chgrp ossec {} \; > /dev/null 2>&1
```
Comment (TomasTurina) [inline debs/SPECS/wazuh-agent/debian/postrm:71]: Some files have root group and/or user by default in a fresh install. Can you validate that this change allows Wazuh to execute all its functionalities?

File: wpk/run.sh [hunk 3]
```diff
@@ -85,6 +86,10 @@ main() {
                     HAVE_PKG_NAME_WIN=true
                 elif [ "${PKG_NAME: -4}" == ".pkg" ]; then
                     HAVE_PKG_NAME_MAC=true
+                elif [ "${PKG_NAME: -4}" == ".rpm" ]; then
```
Comment (TomasTurina) [inline wpk/run.sh:89]: What about APK?

Comment (jr0me) [reply wpk/run.sh:89]: Added in ed90d7125ea40a260c7764955ea6dbdba6f1a759

File: debs/SPECS/wazuh-agent/debian/postrm (hunk 2 above)
Comment (jr0me) [reply debs/SPECS/wazuh-agent/debian/postrm:71]: These lines take files that are _not_ owned by root (user nor group) and assigns them to ossec. These is only run in downgrades to older Wazuh versions.

File: debs/SPECS/wazuh-agent/debian/postrm (hunk 1 above)
Comment (jr0me) [reply debs/SPECS/wazuh-agent/debian/postrm:60]: I've updated the script to also delete the `sockets` folder.

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**